content/gsoc/2025/_index.md

```diff
@@ -52,11 +54,54 @@ The type system design will require particular attention to several key areas. T
+
+### Technical details
+
+`pytest` includes several beneficial features, such as parameterization of tests and using fixtures. There are multiple ongoing efforts to utilize `pytest`'s features better within PyBaMM ([parameterization and fixtures](https://github.com/pybamm-team/PyBaMM/issues/4502), [removing shared.py](https://github.com/pybamm-team/PyBaMM/pull/4401), [more fixtures](https://github.com/pybamm-team/PyBaMM/issues/4837)), but we still have a long way to go. The project will start by picking up existing issues and cherry-picking PRs that are working towards a better testing suite, then slowly take over to create custom `pytest` fixtures or even plugins if required. The mentee will also be tasked to update the deprecated/outdated libraries and syntax used in the testing suite, such as `numpy.assert_almost_equal` statements. Therefore, the project requires auditing the entire testing suite, pointing out outdated, redundant, or deprecated tools and libraries being used, and replacing them with more modern (or `pytest`) solutions.
+
+### Expected outcomes
+
+- Efficient tests: Tests that act exactly how they act right now, but are better organised and use the right tool for the right job (instead of using an arbitrary tool that does the job)
+
+- Implementation: Refactored tests using `pytest` plugins and features; audit of the testing suite; no outdated syntax remaining
+
+- Validation: CI passes on each pull request and the coverage does not go down; tests look more maintainable for the longer run (that is, they are easier to extend or debug)
+
+- Documentation: Updated contributing guide to reflect the new testing conventions for future developers
+
+### Desired skills
+
+- Some experience with unit testing using `pytest` in Python
+- Python programming experience, Git version control, and GitHub workflow for open-source projects
+- An affinity for reading lots of code
+- As a plus, knowledge on how to use scientific computing libraries (NumPy, SciPy)
+- Some experience with using `pytest` plugins is beneficial, but not required
+- Some understanding of continuous integration providers (GitHub Actions, etc.) and coverage tools (Codecov, `pytest-cov`, etc.) is beneficial, but not required
+
+### Difficulty
+
+**Easy**. This project is suitable for a 175-hour duration.
+
+### Potential mentors
+
+* [Saransh Chopra](https://Saransh-cpp.github.io/)
+
+<!-- * [Pradyot Ranjan](https://github.com/prady0t/)
+* [Agriya Khetarpal](https://github.com/agriyakhetarpal/)
+* [Valentin Sulzer](https://github.com/valentinsulzer)
+* [Robert Timms](https://github.com/rtimms)
+* [Arjun Verma](https://arjxnpy.vercel.app/)
+* [Ferran Brosa Planella](https://www.brosaplanella.xyz/) -->
```

Comment:
Feel free to uncomment your names!

Reply:
I can think about it when we will have finalised the projects in the upcoming month(s). :)